warehouse/accounts/services.py

```diff
@@ -561,17 +561,24 @@ def __init__(self, secret, salt, max_age):
     def dumps(self, data):
         return self.serializer.dumps({key: str(value) for key, value in data.items()})
 
-    def loads(self, token):
+    def loads(self, token, return_timestamp=False):
         if not token:
             raise TokenMissing
 
         try:
-            data = self.serializer.loads(token, max_age=self.max_age)
+            if return_timestamp:
```

Comment:
This could probably just be ``data = self.serializer.loads(token, max_age=self.max_age, return_timestamp=return_timestamp)``. It'll return a tuple in return timestamp cases, and a single value in not.

Reply:
ah, and just return `data` to the caller. yeah that's probably tidier.